core/src/main/java/org/geysermc/geyser/item/GeyserCustomItemData.java

```diff
@@ -162,6 +175,12 @@ public Builder renderOffsets(CustomRenderOffsets renderOffsets) {
             return this;
         }
 
+        @Override
+        public Builder tags(Set<String> tags) {
+            this.tags.addAll(tags);
+            return this;
+        }
```

Comment:
Typically with builders, calling the same method a second time will override any changes the first call had.

Especially since this is called `tags`, not `appendTags`, I would suggest making this method clear the existing set.

Also, the nullability annotations should be in the implementation as well. The `tags` parameter may be null as per the API, so that must be taken into consideration as well.

Reply:
https://github.com/GeyserMC/Geyser/blob/f2614bca8625437318924d6d9c1bc2959d805729/core/src/main/java/org/geysermc/geyser/level/block/GeyserCustomBlockComponents.java#L294-L298
I basically copied how tags are added for custom blocks; can revert it - not sure whether to leave tags nullable (as is right now), or to make them NotNull, same as tags are implemented for custom blocks.

Reply:
ah. this is out of scope of my review then

Reply:
https://github.com/GeyserMC/Geyser/pull/4041/commits/d3d4b494235846ca772ae0d7012b329e5f218eb4
Fixed here - tags for items are now, just as tags for blocks, marked NonNull. The builder for both was changed to set the tags, not to add to them.